lib/streamlit/elements/image.py

```diff
@@ -94,6 +99,8 @@ def image(
         clamp: bool = False,
         channels: Channels = "RGB",
         output_format: ImageFormatOrAuto = "auto",
+        *,
+        use_container_width: bool | None = None,
```

Comment:
Is there a good reason why we should not just make it `use_container_width: bool = False`? I believe we would be able to remove [these lines](https://github.com/streamlit/streamlit/pull/9547/files#diff-4795158725eb992f73736495442ff69edd6e7ccd7609291339a61d1b6e39d716R192-R194) then.

Reply:
I think that would make sense if we weren't also trying to deprecate the `use_column_width` arg. If this is set this to `False` by default then it can't also throw an API error in the cases where the user has explicitly used both args [here](https://github.com/streamlit/streamlit/pull/9547/files#diff-4795158725eb992f73736495442ff69edd6e7ccd7609291339a61d1b6e39d716R170-R174)

Reply:
Ahhh ok, yeah that makes sense! Though we could change raising the error to the following scenario where `use_column_width` is not `None` and `use_container_width` is set to `True`.
And then always show a deprecation warning whenever `use_column_width` is used by leveraging the `deprecation_util.show_deprecation_warning` function. This is shown only in dev mode to the developers and I think how we usually go about deprecations. Then we can keep the API here clean and just use `bool`.
How do you feel about this?

Reply:
Thanks for your thoughtful comments here. After doing some iterations, I agree with the suggestions here and have updated the PR to reflect this.